extensions/amp-social-share/0.2/social-share.js

```diff
@@ -0,0 +1,188 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Keys} from '../../../src/utils/key-codes';
+import {Services} from '../../../src/services';
+import {addParamsToUrl, parseQueryString} from '../../../src/url';
+import {createElement, useContext, useState} from '../../../src/preact';
+import {dev, devAssert, user, userAssert} from '../../../src/log';
+import {dict} from '../../../src/utils/object';
+import {getAmpContext} from '../../../src/preact/context';
+import {getSocialConfig} from './amp-social-share-config';
+import {openWindowDialog} from '../../../src/dom';
+import {useResourcesNotify} from '../../../src/preact/utils';
+
+const socialShareStyle = `background-repeat: no-repeat;
+  background-position: center;
+  background-size: contain;
+  text-decoration: none;
+  cursor: pointer;
+  position: relative;
+  background-color: black; 
+  background-image: url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%20512%20512%22%3E%3Cpath%20fill%3D%22%23ffffff%22%20d%3D%22M211.9%20197.4h-36.7v59.9h36.7V433.1h70.5V256.5h49.2l5.2-59.1h-54.4c0%200%200-22.1%200-33.7%200-13.9%202.8-19.5%2016.3-19.5%2010.9%200%2038.2%200%2038.2%200V82.9c0%200-40.2%200-48.8%200%20-52.5%200-76.1%2023.1-76.1%2067.3C211.9%20188.8%20211.9%20197.4%20211.9%20197.4z%22%3E%3C%2Fpath%3E%3C%2Fsvg%3E');
+`;
+
+/**
+ * @param {!JsonObject} props
+ * @return {Preact.Renderable}
+ */
+export function SocialShare(props) {
+  const viewer = Services.viewerForDoc(props['host']);
+  const platform = Services.platformFor(window);
+  const typeConfig = getTypeConfigOrUndefined(props['type'], viewer, platform);
+  // Hide/ignore component if typeConfig is undefined
+  if (!typeConfig) {
+    const {'collapse': collapse} = useContext(getAmpContext());
+    return collapse();
+  }
+  const shareEndpoint = userAssert(
+    props['dataShareEndpoint'] || typeConfig['shareEndpoint'],
+    'The dataShareEndpoint property is required. %s'
+  );
+  const urlParams = {...typeConfig['defaultParams'], ...props['dataParams']};
+  const hrefWithVars = addParamsToUrl(
+    dev().assertString(shareEndpoint),
+    urlParams
+  );
+  const urlReplacements = Services.urlReplacementsForDoc(props['host']);
+  const bindingVars = typeConfig['bindings'];
+  const bindings = {};
+  if (bindingVars) {
+    bindingVars.forEach(name => {
+      const bindingName = name.toUpperCase();
+      bindings[bindingName] = urlParams[name];
+    });
+  }
+
+  const {0: href, 1: setHref} = useState();
+  const {0: target, 1: setTarget} = useState();
+
+  urlReplacements.expandUrlAsync(hrefWithVars, bindings).then(result => {
```

Comment:
This seems highly AMP specific. Wouldn't it be better to expose just a `href` prop, and we use it without modification?

Reply:
Good point, done.